packages/SwingSet/misc-tools/store-stats/src/store-stats.js

```diff
@@ -0,0 +1,160 @@
+    findHeight: () => {
+      console.log('Fetching block height from the database');
+      return kvGet('host.height');
+    },
+    findVatItems: () => {
+      console.log('Fetching vat items from the database');
+      return kvAll('v[0-9]*.*');
+    },
+  });
+};
+
+export const makeStatsSender = () => {
+  const sendMetrics = data => {
+    const { height, vatStats } = data;
+    console.log(
+      `Sending metrics of ${vatStats.size} vats to datadog for block ${height}`,
+    );
+    for (const [vatID, stats] of vatStats) {
+      const tags = [`height:${height}`, `vatID:${vatID}`];
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_exports_reachable',
+        stats.objectExportsReachable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_exports_recognizable',
+        stats.objectExportsRecognizable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_imports_reachable',
+        stats.objectImportsReachable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_imports_recognizable',
+        stats.objectImportsRecognizable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.promises',
+        stats.promises,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.vat_store_keys',
+        stats.vatStoreKeys,
+        tags,
+      );
+      dogstatsd.increment('swingset.kvstore.stats');
+    }
+    dogstatsd.close();
+    console.log('Done sending metrics to datadog');
+  };
+
+  return Object.freeze({
+    sendMetrics,
+  });
+};
+
+export const makeStatsTransformer = () => {
+  // kvstore has pairs like:
+  // "v$NN.c.ko45": "R o+4"      # reachable export
+  // "v$NN.c.ko46": "_ o+5"      # merely-recognizable export
+  // "v$NN.c.ko47": "R o-6"      # reachable import
+  // "v$NN.c.ko48": "_ o-7"      # merely-recognizable import
+  // "v$NN.c.kp70": "R p+12"      # promise (always "R", sometimes "p+" sometimes "p-")
+
+  const object = /^v\d+\.c\..*ko\d.*$/; // vNN.c.koNNNNN*
+  const exportReach = /^R o\+.*$/; // R o+*
+  const exportRecog = /^_ o\+.*$/; // _ o+*
+
+  const importReach = /^R o-.*$/; // R o-*
+  const importRecog = /^_ o-.*$/; // _ o-*
+
+  const vatPromise = /^v\d+\.c\..*kp\d.*$/; // vNN.c.kpNNNNN*
+  const vatStoreKeys = /^v\d+\.vs\..*$/; // vNN.vs.*
+
+  const newStat = () => {
+    return {
+      objectExportsReachable: 0,
+      objectExportsRecognizable: 0,
+      objectImportsReachable: 0,
+      objectImportsRecognizable: 0,
+      promises: 0,
+      vatStoreKeys: 0,
+    };
+  };
+
+  const rowsToStats = rows => {
+    console.log(`Transforming rows to stats`);
+    let count = 0;
+    const vatStats = new Map();
+    for (const row of rows) {
+      count += 1;
+      const vatID = row.key.split('.')[0];
+      const stat = vatStats.get(vatID) || newStat();
+      vatStats.set(vatID, stat);
```

Comment:
since there are a lot of rows, this might be a tiny bit faster if we skip the redundant `.set`, something like
```js
let stat = vatStats.get(vatID);
if (!stat) {
  stat = newStat();
  vatStats.set(vatID, stat);
}
```